docs/usecases/provenance_tracking.rst

```diff
@@ -0,0 +1,293 @@
+       get (notneeded: 1)
+       save (ok: 1)
+
+He continues to process the images, capturing all provenance with DataLad.
+Later, he can always find out which commands produced or changed which file.
+This information is easily accessible within the history of his dataset,
+both with Git and DataLad commands such as :command:`git log` or
+:command:`datalad diff`.
+
+.. code-block:: bash
+
+    $ git log --oneline @~3..@
+     73832b3 (HEAD -> master) [DATALAD RUNCMD] extract pimpernel
+     cce0c79 [DATALAD RUNCMD] extract st-bernard lily
+     8a21b21 Added flower mosaic from wikimedia
+
+    $ datalad diff -f HEAD~3
+        added: pimpernel.jpg (file)
+        added: sources/flowers.jpg (file)
+        added: st-bernard.jpg (file)
+
+Based on this information, he can always reconstruct how an when
+any data file came to be – across the entire life-time of a project.
+
+He decides that one image manipulation for his art project will
+be to displace pixels of an image by a random amount to blur the image:
+
+.. code-block:: bash
+
+    $ datalad run -m "blur image" \
+      --input "st-bernard.jpg" \
+      --output "st-bernard-displaced.jpg" \
+      "convert -spread 10 st-bernard.jpg st-bernard-displaced.jpg"
+
+      [INFO   ] Making sure inputs are available (this may take some time)
+      [INFO   ] == Command start (output follows) =====
+      [INFO   ] == Command exit (modification check follows) =====
+         add(ok): st-bernard-displaced.jpg (file)
+         save(ok): /demo/demo (dataset)
+         action summary:
+           add (ok: 1)
+           get (notneeded: 1)
+           save (ok: 1)
+
+Because he is not completely satisfied with the first random pixel displacement,
+he decides to retry the operation. Because everything was wrapped in :command:`datalad run`,
+he can rerun the command. The command will produce a commit, because the displacement is
+random and the output file changes slightly from its previous version.
+
+
+.. code-block:: bash
+
+   $ git log -1 --oneline HEAD
+     643c175 (HEAD -> master) [DATALAD RUNCMD] blur image
+   # rerun with the commit hash:
+   $ datalad rerun 643c175
+
+   [INFO   ] Making sure inputs are available (this may take some time)
+   unlock(ok): st-bernard-displaced.jpg (file)
+   [INFO   ] == Command start (output follows) =====
+   [INFO   ] == Command exit (modification check follows) =====
+   add(ok): st-bernard-displaced.jpg (file)
+   save(ok): . (dataset)
+   action summary:
+     add (ok: 1)
+     get (notneeded: 1)
+     save (ok: 1)
+     unlock (ok: 1)
+
+This blur also does not yet fulfill Robs expectations, so he decides to
+discard the change, using standard Git tools [#f3]_.
+
+.. code-block:: bash
+
+   $ git reset --hard HEAD~1
+     HEAD is not at 643c175 [DATALAD RUNCMD] blur image
+
+He knows that within a DataLad dataset, he can also rerun multiple commands
+with ``--since``  and ``--onto`` to specify where to start rerunning from and
+up to which point.
```

Comment:
This sounds off.  I'll describe it in git terms and let you translate as you think is appropriate for this page.

A call `rerun --since=A --onto B REV` reruns the run commits in the range `A..REV` with `HEAD` set to `B` (i.e., `B` is the new base).  It is very much like `git rebase`s `--onto`.

As an added complication, empty strings for --since and --onto are special cases and result in replaying all the run commits from the current branch in another branch.